comp/logs/agent/config/config.go

```diff
@@ -172,6 +172,36 @@ func buildTCPEndpoints(coreConfig pkgconfigmodel.Reader, logsConfig *LogsConfigK
 	}
 
 	additionals := loadTCPAdditionalEndpoints(main, logsConfig)
+
+	// Add in the MRF endpoint if MRF is enabled.
+	if coreConfig.GetBool("multi_region_failover.enabled") {
+		mrfURL, err := pkgconfigutils.GetMRFLogsEndpoint(coreConfig, tcpEndpointPrefix)
+		if err != nil {
+			return nil, fmt.Errorf("cannot construct MRF endpoint: %s", err)
+		}
+
+		mrfHost, mrfPort, mrfUseSSL, err := parseAddressWithScheme(mrfURL, defaultNoSSL, parseAddressAsHost)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse %s: %v", mrfURL, err)
+		}
+
+		e := NewEndpoint(coreConfig.GetString("multi_region_failover.api_key"), "multi_region_failover.api_key", mrfHost, mrfPort, mrfUseSSL)
+		e.IsMRF = true
+		e.UseCompression = main.UseCompression
+		e.CompressionLevel = main.CompressionLevel
+		e.BackoffBase = main.BackoffBase
+		e.BackoffMax = main.BackoffMax
+		e.BackoffFactor = main.BackoffFactor
+		e.RecoveryInterval = main.RecoveryInterval
+		e.RecoveryReset = main.RecoveryReset
+		e.Version = main.Version
+		e.TrackType = intakeTrackType
+		e.Protocol = intakeProtocol
+		e.Origin = intakeOrigin
```

Comment:
I don't think TrackType, protocol, or Origin are used in TCP mode. 
If this is the case, you can omit these.

Reply:
Sounds good. They're included in [`loadTCPAdditionalEndpoints()`](https://github.com/DataDog/datadog-agent/blob/main/comp/logs/agent/config/endpoints.go#L149), is that also an anomaly?

Reply:
They are copied from the main endpoint, but the main endpoint never sets them. So I don't think they are ever actually used.